app/src/main/java/net/vonforst/evmap/api/nobil/NobilApi.kt

```diff
@@ -0,0 +1,332 @@
+package net.vonforst.evmap.api.nobil
+
+import android.content.Context
+import android.database.DatabaseUtils
+import com.car2go.maps.model.LatLng
+import com.car2go.maps.model.LatLngBounds
+import com.squareup.moshi.JsonDataException
+import com.squareup.moshi.Moshi
+import net.vonforst.evmap.BuildConfig
+import net.vonforst.evmap.R
+import net.vonforst.evmap.addDebugInterceptors
+import net.vonforst.evmap.api.ChargepointApi
+import net.vonforst.evmap.api.ChargepointList
+import net.vonforst.evmap.api.FiltersSQLQuery
+import net.vonforst.evmap.api.FullDownloadResult
+import net.vonforst.evmap.api.StringProvider
+import net.vonforst.evmap.api.mapPower
+import net.vonforst.evmap.api.mapPowerInverse
+import net.vonforst.evmap.api.powerSteps
+import net.vonforst.evmap.model.BooleanFilter
+import net.vonforst.evmap.model.ChargeLocation
+import net.vonforst.evmap.model.Chargepoint
+import net.vonforst.evmap.model.ChargepointListItem
+import net.vonforst.evmap.model.Filter
+import net.vonforst.evmap.model.FilterValue
+import net.vonforst.evmap.model.FilterValues
+import net.vonforst.evmap.model.MultipleChoiceFilter
+import net.vonforst.evmap.model.ReferenceData
+import net.vonforst.evmap.model.SliderFilter
+import net.vonforst.evmap.model.getBooleanValue
+import net.vonforst.evmap.model.getMultipleChoiceValue
+import net.vonforst.evmap.model.getSliderValue
+import net.vonforst.evmap.viewmodel.Resource
+import okhttp3.Cache
+import okhttp3.OkHttpClient
+import retrofit2.HttpException
+import retrofit2.Response
+import retrofit2.Retrofit
+import retrofit2.converter.moshi.MoshiConverterFactory
+import retrofit2.http.Body
+import retrofit2.http.GET
+import retrofit2.http.POST
+import retrofit2.http.Query
+import java.io.IOException
+import java.time.Duration
+
+private const val maxResults = 2000
+
+interface NobilApi {
+    @GET("datadump.php")
+    suspend fun getAllChargingStations(
+        @Query("apikey") apikey: String,
+        @Query("countrycode") countrycode: String = "DAN",
```

Comment:
This line should be removed. It's only there to bypass a NetworkOnMainThreadException I can't figure out.